### DISCUSSION TITLE
fluxcd msteams provider not able to use secret ?

### DISCUSSION THREAD

**Discussion by mihaitacristian** [fluxcd/flux2#2914] [Q&A]
I'm trying to use VaultSecret or a generic secret in order to add the secret on provider msteams configuration. The problem is that the msteam-provider is not able to initialize it.  

![secret](https://user-images.githubusercontent.com/51115257/179390975-6577b525-e0c8-4e30-a519-def7a3ea15ed.png)
![provider](https://user-images.githubusercontent.com/51115257/179390980-511dbfe0-4be1-4815-bfaa-b3a090e9e275.PNG)

**Comment by pjbgf** [comment]
I haven't used `VaultSecret` before, but I am assuming that it is working as expected and error-free.

Have you tried checking whether `VaultSecret`  is creating the target secret and it contains the `address` field as expected by the Notification Controller?
If that is still not working, would you be able to share any error messages or the logs from your Notification Controller?
